### PR TITLE
add entry point to structure_manager_widget

### PR DESCRIPTION
**Pull request by superstar54**
Continue the plugin implementation. This PR adds an entry point for step 1,  allowing the user to add a new structure importer and editor.

In principle, all the structure `importer` and `editor` are pluggable. This PR just generalizes this by adding entry point to it.

This will be very useful. 
- add structure importer specific for particular structures, e.g. surface,  adsorbate.
- add a new `editor` to edit a structure for the plugin, e.g. edit tags, and cut surface. Like the `AddingTagsEditor` by @AndresOrtegaGuerrero

### CODE REVIEW

**Comment by AndresOrtegaGuerrero** [comment]
@superstar54 I noticed with this PR , I can reach my StructureData, it says i have 0.

**Comment by AndresOrtegaGuerrero** [comment]
@superstar54 I guess we need a sample pluggin to test this PR right ? in such a way that the pluggin includes an "extra" step 1 ?

File: src/aiidalab_qe/app/structure/__init__.py [hunk 1]
```diff
@@ -22,15 +34,43 @@
     ("Cobalt (hcp)", file_path / "examples" / "Co.cif"),
 ]
 
+OptimadeQueryWidget.title = "OPTIMADE"  # monkeypatch
+
 
 class StructureSelectionStep(ipw.VBox, WizardAppWidgetStep):
     """Integrated widget for the selection of structures from different sources."""
```
Comment (AndresOrtegaGuerrero) [inline src/aiidalab_qe/app/structure/__init__.py:41]: I feel the Docstring in this class should be updated to incorporate the structure of the class, and also a description of modification of the importer and editor

**Comment by AndresOrtegaGuerrero** [comment]
@superstar54 For example in the example for testing if I have 
class EosOutline(OutlinePanel):
    title = "Equation of State (EOS)"


eos ={
"outline": EosOutline,
"importer": EOSStructure,
"setting": Setting,
"workchain": workchain_and_builder,
"result": Result,
}

i can just declare something like  "editor": EOSEditor, should it work ?

File: src/aiidalab_qe/app/structure/__init__.py [hunk 2]
```diff
@@ -22,15 +34,43 @@
...
+            importers.append(entry_point())
+        #
+        editors = [
+            BasicCellEditor(title="Edit cell"),
+            BasicStructureEditor(title="Edit structure"),
+            AddingTagsEditor(title="Edit StructureData"),
+        ]
+        entries = get_entry_items("aiidalab_qe.properties", "editor")
+        for _, entry_point in entries.items():
+            importers.append(entry_point())
```
Comment (AndresOrtegaGuerrero) [inline src/aiidalab_qe/app/structure/__init__.py:65]: should it be here `editors.append(entry_point())`

File: src/aiidalab_qe/app/structure/__init__.py (hunk 1 above)
Comment (superstar54) [reply src/aiidalab_qe/app/structure/__init__.py]: Thanks, I have updated it.

File: src/aiidalab_qe/app/structure/__init__.py (hunk 2 above)
Comment (superstar54) [reply src/aiidalab_qe/app/structure/__init__.py]: Yes, I changed it. thanks!

**Comment by superstar54** [comment]
Hi @AndresOrtegaGuerrero , thanks for the review. I have made the changes as you suggested.

> i can just declare something like "editor": EOSEditor, should it work ?

Yes, you can add the editor like this. I have update the `aiidalab_qe_eos`, you can test the `importer` and `editor` use it.

File: src/aiidalab_qe/app/structure/__init__.py [hunk 3]
```diff
@@ -22,15 +34,47 @@
...
+        entries = get_entry_items("aiidalab_qe.properties", "importer")
+        for _, entry_point in entries.items():
+            importers.append(entry_point())
+        # add plugin specific structure editors
+        editors = [
+            BasicCellEditor(title="Edit cell"),
+            BasicStructureEditor(title="Edit structure"),
+            AddingTagsEditor(title="Edit StructureData"),
+        ]
+        entries = get_entry_items("aiidalab_qe.properties", "editor")
```
Comment (AndresOrtegaGuerrero) [inline src/aiidalab_qe/app/structure/__init__.py:67]: what about list comprehension like `importers.extend(entry_point() for entry_point in get_entry_items("aiidalab_qe.properties", "importer").values())` , this is not important just wondering :)

**Comment by AndresOrtegaGuerrero** [comment]
@superstar54 Thank you again! I tried the PR it works, but it seems some of the test didnt pass

**Comment by superstar54** [comment]
@AndresOrtegaGuerrero Thanks. I fixed it. All tests pass now.

**Review by AndresOrtegaGuerrero — APPROVED** [review]
LGTM! Check my comment on the list comprehension , in case you think is useful

File: src/aiidalab_qe/app/structure/__init__.py (hunk 3 above)
Comment (superstar54) [reply src/aiidalab_qe/app/structure/__init__.py:66]: thanks for the suggestion. I used the list comprehension.

**Pull request merged**